### ISSUE TITLE
Akka.Remote EndpointManager error

### ISSUE THREAD

**Issue by skotzko** [akkadotnet/akka.net#1052]
In testing a refactor of an [Akka.Cluster demo](https://github.com/petabridge/akkadotnet-code-samples/tree/master/Cluster.WebCrawler), ran into the below issue w/ the EndpointManager. At the time it occurred, there was a lot of disconnecting/reconnecting.

### Stack trace
```
[WARNING][6/11/2015 1:09:07 AM][Thread 0009][remoting] Association to [akka.tcp:
//webcrawler@127.0.0.1:4053] having UID [1324626967] is irrecoverably failed. UI
D is now quarantined and all messages to this UID will be delivered to dead lett
ers. Remote actorsystem must be restarted to recover from this situation.
[WARNING][6/11/2015 1:09:07 AM][Thread 0009][remoting] Association to [akka.tcp:
//webcrawler@127.0.0.1:17241] having UID [1899179539] is irrecoverably failed. U
ID is now quarantined and all messages to this UID will be delivered to dead let
ters. Remote actorsystem must be restarted to recover from this situation.
[WARNING][6/11/2015 1:09:09 AM][Thread 0011][[akka://webcrawler/system/endpointM
anager/reliableEndpointWriter-akka.tcp%3a%2f%2fwebcrawler%40127.0.0.1%3a17251-9]
] Association with remote system akka.tcp://webcrawler@127.0.0.1:17251 has faile
d; address is now gated for 5000 ms. Reason is: [Disassociated]
[ERROR][6/11/2015 1:09:09 AM][Thread 0011][akka://webcrawler/system/endpointMana
ger/reliableEndpointWriter-akka.tcp%3a%2f%2fwebcrawler%40127.0.0.1%3a17251-9/end
pointWriter] Disassociated
Cause: Akka.Remote.EndpointDisassociatedException: Disassociated
   at Akka.Remote.EndpointWriter.PublishAndThrow(Exception reason, LogLevel leve
l)
   at Akka.Remote.EndpointWriter.Unhandled(Object message)
   at Akka.Remote.EndpointWriter.Writing(Object message)
   at Akka.Actor.ActorCell.<>c__DisplayClass3e.<Akka.Actor.IUntypedActorContext.
Become>b__3d(Object m)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Akka.Actor.ActorCell.ReceiveMessage(Object message)
   at Akka.Actor.ActorCell.ReceivedTerminated(Terminated t)
   at Akka.Actor.ActorCell.AutoReceiveMessage(Envelope envelope)
   at Akka.Actor.ActorCell.Invoke(Envelope envelope)
[ERROR][6/11/2015 1:09:09 AM][Thread 0005][akka://webcrawler/system/endpointMana
ger] The given key was not present in the dictionary.
Cause: System.Collections.Generic.KeyNotFoundException: The given key was not pr
esent in the dictionary.
   at System.Collections.Generic.Dictionary`2.get_Item(TKey key)
   at Akka.Remote.EndpointRegistry.RegisterWritableEndpointUid(IActorRef writer,
 Int32 uid)
   at Akka.Remote.EndpointManager.<Accepting>b__29(GotUid gotuid)
   at Akka.Case.With[TMessage](Action`1 action)
   at Akka.Remote.EndpointManager.Accepting(Object message)
   at Akka.Actor.ActorCell.<>c__DisplayClass3e.<Akka.Actor.IUntypedActorContext.
Become>b__3d(Object m)
   at Akka.Actor.ActorBase.AroundReceive(Receive receive, Object message)
   at Akka.Actor.ActorCell.ReceiveMessage(Object message)
   at Akka.Actor.ActorCell.Invoke(Envelope envelope)

```

**Comment by Aaronontheweb** [comment]
Possibly related to #1147

**Comment by skotzko** [comment]
Closing as this was previously fixed in #1147.
